app/controllers/katello/api/v2/module_streams_controller.rb

```diff
@@ -3,8 +3,43 @@ class Api::V2::ModuleStreamsController < Api::V2::ApiController
     apipie_concern_subst(:a_resource => N_("a module stream"), :resource => "module_streams")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :check_hosts, :only => :index
+
+    api :GET, "/module_streams", N_("List packages")
+    api :GET, "/content_views/:content_view_id/filters/:filter_id/:resource_id", N_("List :resource_id")
+    api :GET, "/content_view_filters/:content_view_filter_id/:resource_id", N_("List :resource_id")
+    api :GET, "/repositories/:repository_id/:resource_id", N_("List :resource_id")
+
+    param :organization_id, :number, :desc => N_("organization identifier")
+    param :content_view_version_id, :number, :desc => N_("content view version identifier")
+    param :content_view_filter_id, :number, :desc => N_("content view filter identifier")
+    param :repository_id, :number, :desc => N_("repository identifier")
+    param :environment_id, :number, :desc => N_("environment identifier")
+    param :ids, Array, :desc => N_("ids to filter content by")
+    param :host_ids, Array, :desc => N_("List of host id to list available module streams for")
+    param_group :search, ::Katello::Api::V2::ApiController
+    def index
+      super
+    end
```

Comment:
Do you need this method? I think you can just define custom_index_relation

Reply:
I guess you are doing this to add `host_ids` param?

Reply:
yup I needed to add the host_ids param. Thats the only reason